anesthetic/samples.py

```diff
@@ -264,12 +250,11 @@ def plot_2d(self, axes, *args, **kwargs):
                          'upper': types[-1]}
             else:
                 types = {'lower': types[0], 'upper': types[-1]}
-        diagonal_kwargs = kwargs.pop('diagonal_kwargs', {})
-        lower_kwargs = kwargs.pop('lower_kwargs', {})
-        upper_kwargs = kwargs.pop('upper_kwargs', {})
-        diagonal_kwargs.update(kwargs)
-        lower_kwargs.update(kwargs)
-        upper_kwargs.update(kwargs)
+
+        local_kwargs = {pos: kwargs.pop('%s_kwargs' % pos, {})
+                        for pos in ['lower', 'upper', 'diagonal']}
```

Comment:
lol, and we've got our dict in dict after all... at least we are sparing the user of having to deal with it...